lib/cart/apis/cart_exceptions.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'cart_exceptions.g.dart';
+
+enum CartExceptionReason {
+  notFound,
+}
+
+@JsonSerializable(createToJson: false)
+class CartException implements Exception {
+  final String message;
+  final CartExceptionReason reason;
+  final String subject;
+  final String subjectType;
+  final String type;
```

Comment:
Those non-nullable fields make the parsing of the exception tightly coupled to what Hybris is sending to us.

Example: if there is no `subject` in the response for a reason or another, the parsing will fail with a NPE.

Reply:
I was thinking the same thing, but had the same principle "deal with poor exceptions when we face them". Thoughts?

Reply:
Yep, let's see 😎